csharp/src/Ice/Reference.cs

```diff
@@ -168,8 +168,10 @@ public static Reference Parse(string s, Communicator communicator, string? prope
                 }
 
                 property = $"{propertyPrefix}.InvocationTimeout";
-                invocationTimeout = communicator.GetPropertyAsInt(property);
-                if (invocationTimeout != null && invocationTimeout < 1 && invocationTimeout != -1)
+                invocationTimeout = communicator.GetPropertyAsTimeSpan(property);
```

Comment:
Same here we can remove this check `GetPropertyAsTimeSpan` will never return a negative value other than `Timeout.InfiniteTimeSpan`

Reply:
Unlike other timeouts, we don't accept 0 invocation timeouts. So we need a check to reject 0 timeouts.